core/src/test/java/io/questdb/test/metrics/QueryTracingSubstitutionTest.java

```diff
@@ -0,0 +1,160 @@
+import io.questdb.std.str.StringSink;
+import io.questdb.test.AbstractBootstrapTest;
+import io.questdb.test.TestServerMain;
+import org.junit.Assert;
+import org.junit.Test;
+import org.postgresql.util.PSQLException;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static io.questdb.test.cutlass.pgwire.BasePGTest.assertResultSet;
+import static io.questdb.test.cutlass.pgwire.BasePGTest.printToSink;
+import static io.questdb.test.tools.TestUtils.assertMemoryLeak;
+
+@SuppressWarnings("SqlDialectInspection")
+public class QueryTracingSubstitutionTest extends AbstractBootstrapTest {
+
+    @Test
+    public void testSubstitutionOfBindVariables() throws Exception {
+        assertMemoryLeak(() -> {
+            createDummyConfiguration("pg.select.cache.enabled=true", "query.tracing.enabled=true", "query.tracing.bind.variable.substitution.enabled=true");
+            try (final ServerMain serverMain = TestServerMain.createWithManualWalRun(getServerMainArgs())) {
+                serverMain.start();
+
+                try (Connection connection = getConnection(serverMain)) {
+                    try (final PreparedStatement stmt = connection.prepareStatement("SELECT ?, ?, ?, ?, ?, ?, ?, ?;")) {
+                        stmt.setBoolean(1, true);
+                        stmt.setByte(2, (byte) 111);
+                        stmt.setDouble(3, 123.456);
+                        stmt.setFloat(4, (float) 123.456);
+                        stmt.setInt(5, 987654);
+                        stmt.setLong(6, 987654L);
+                        stmt.setShort(7, (short) 11111);
+                        stmt.setString(8, "test");
+
+                        try (final ResultSet resultSet = stmt.executeQuery()) {
+                            assertResultSet(
+                                    "$1[BIT],$2[SMALLINT],$3[DOUBLE],$4[REAL],$5[INTEGER],$6[BIGINT],$7[SMALLINT],$8[VARCHAR]\n" +
+                                            "true,111,123.456,123.456,987654,987654,11111,test\n",
+                                    Misc.getThreadLocalSink(),
+                                    resultSet
+                            );
+                        }
+                    }
+
+                    int sleepMillis = 100;
+                    while (true) {
+                        //noinspection BusyWait
+                        Thread.sleep(sleepMillis);
+                        try {
+                            try (final PreparedStatement stmt = connection.prepareStatement("_query_trace;")) {
+                                try (final ResultSet resultSet = stmt.executeQuery()) {
+                                    StringSink sink = Misc.getThreadLocalSink();
+                                    printToSink(sink, resultSet, null);
+                                    Assert.assertFalse(sink.toString().contains("$"));
```

Comment:
i would compare the full content of the sink with the expected result instead of just checking that it does not contain the `$` sign.